components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -3503,6 +3508,128 @@ private void deleteRequestPathAuthenticators(int applicationID, Connection conne
+     */
+    private SpTrustedAppMetadata getTrustedAppMetadata(int applicationId, Connection connection, int tenantID)
+            throws IdentityApplicationManagementException {
+
+        PreparedStatement loadTrustedAppConfigs = null;
+        ResultSet trustedAppConfigResultSet = null;
+        SpTrustedAppMetadata trustedAppMetadata = new SpTrustedAppMetadata();
+
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving trusted app configurations for Application " + applicationId);
+        }
+        try {
+            loadTrustedAppConfigs = connection
+                    .prepareStatement(ApplicationMgtDBQueries.LOAD_TRUSTED_APPS_BY_APP_ID);
+            loadTrustedAppConfigs.setInt(1, applicationId);
+            loadTrustedAppConfigs.setInt(2, tenantID);
+            trustedAppConfigResultSet = loadTrustedAppConfigs.executeQuery();
+
+            while (trustedAppConfigResultSet.next()) {
+                String platformType = trustedAppConfigResultSet.getString(1);
+                if (ANDROID.equals(platformType)) {
+                    trustedAppMetadata.setAndroidPackageName(trustedAppConfigResultSet.getString(2));
+                    trustedAppMetadata.setAndroidThumbprints(trustedAppConfigResultSet.getString(3));
+                } else if (IOS.equals(platformType)) {
+                    trustedAppMetadata.setAppleAppId(trustedAppConfigResultSet.getString(2));
+                }
+                trustedAppMetadata.setIsFidoTrusted(trustedAppConfigResultSet.getBoolean(4));
+                trustedAppMetadata.setIsTWAEnabled(trustedAppConfigResultSet.getBoolean(5));
+            }
+        } catch (SQLException e) {
+            throw new IdentityApplicationManagementException("Error while retrieving trusted app configurations.", e);
+        } finally {
+            IdentityApplicationManagementUtil.closeStatement(loadTrustedAppConfigs);
+            IdentityApplicationManagementUtil.closeResultSet(trustedAppConfigResultSet);
+        }
+        return trustedAppMetadata;
+    }
+
+    /**
+     * Update trusted app metadata of the application.
+     *
+     * @param applicationId  Application ID.
+     * @param trustedAppMetadata Trusted app configurations.
+     * @param connection     Database connection.
+     * @throws IdentityApplicationManagementException If an error occurs while updating trusted app configurations.
+     */
+    private void updateTrustedAppMetadata(int applicationId, SpTrustedAppMetadata trustedAppMetadata,
+                                          Connection connection) throws IdentityApplicationManagementException {
+
+        int tenantID = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        PreparedStatement storeTrustedAppConfigs = null;
+
+        if (log.isDebugEnabled()) {
+            log.debug("Adding trusted app configurations for Application " + applicationId);
```

Comment:
log.debug("Adding trusted app configurations for application: " + applicationId);